test/fixtures/rules/std_rule_cases/RF06.yml

```diff
@@ -1,104 +1,106 @@
 rule: RF06
 
-test_pass_column_reference:
-  pass_str: |
-    SELECT 123 AS foo;
-
-test_fail_column_reference:
+test_pass_default:
+  # NOTE: The ANSI dialect casefolds to UPPERCASE.
+  pass_str: |
+    select
+      unquoted_identifier,
+      another as unquoted_alias,
+      1 + 2 as expression_alias,
+      "case_sensitive_reference",
+      "REFERENCE WITH SPACE",
+      "I-c@nn0t-be~un-quoted",
+      "SELECT",  -- Keyword as identifier
+      another as "Case_Sensitive_alias"
+    from my_schema.my_table
+
+test_fail_default_uppercase:
+  # Snowflake is a dialect which casefolds to UPPERCASE
   fail_str: |
-    SELECT 123 AS "foo";
-  fix_str: |
-    SELECT 123 AS foo;
-
-test_fail_column_reference_tsql:
-  fail_str: |
-    SELECT 123 AS [foo];
+    SELECT
+      "COLUMN_REFERENCE",
+      foo AS "COLUMN_ALIAS",
+      "Case_Sensitive_Reference"
+    FROM "SCHEMA_REFERENCE"."TABLE_REFERENCE" AS "TABLE_ALIAS"
   fix_str: |
-    SELECT 123 AS foo;
+    SELECT
+      COLUMN_REFERENCE,
+      foo AS COLUMN_ALIAS,
+      "Case_Sensitive_Reference"
+    FROM SCHEMA_REFERENCE.TABLE_REFERENCE AS TABLE_ALIAS
   configs:
     core:
-      dialect: tsql
-
-test_pass_table_reference:
-  pass_str: |
-    SELECT foo
-    FROM bar;
-
-test_fail_table_reference:
-  fail_str: |
-    SELECT foo
-    FROM "bar";
-  fix_str: |
-    SELECT foo
-    FROM bar;
+      dialect: snowflake
 
-test_fail_table_reference_tsql:
+test_fail_default_lowercase:
+  # Postgres is a dialect which casefolds to lowercase
   fail_str: |
-    SELECT foo
-    FROM [bar];
+    SELECT
+      "column_reference",
+      foo AS "column_alias",
+      "Case_Sensitive_Reference"
+    FROM "schema_reference"."table_reference" AS "table_alias"
   fix_str: |
-    SELECT foo
-    FROM bar;
+    SELECT
+      column_reference,
+      foo AS column_alias,
+      "Case_Sensitive_Reference"
+    FROM schema_reference.table_reference AS table_alias
   configs:
     core:
-      dialect: tsql
-
-test_pass_multiple_references:
-  pass_str: |
-    SELECT foo
-    FROM bar.baz;
-
-test_fail_multiple_references:
-  fail_str: |
-    SELECT foo
-    FROM "bar"."baz";
-  fix_str: |
-    SELECT foo
-    FROM bar.baz;
+      dialect: postgres
 
-test_fail_multiple_references_tsql:
+test_fail_default_insensitive:
+  # DuckDB is a dialect which compares insensitively, which means
+  # as long as the identifier is *valid* without quotes, we should
+  # remove the quotes.
   fail_str: |
-    SELECT foo
-    FROM [bar].[baz];
+    SELECT
+      "lowercase_reference",
+      "UPPERCASE_REFERENCE",
+      foo AS "column_alias",
+      "Case_Sensitive_Reference",
```

Comment:
I guess this should be `Case_Insensitive_Reference`

Reply:
I see that it's ambiguous 🤔 . My intent here was a reference where the case matters, and therefore can't be unquoted. I'll change it to `Case_Specific_Reference`.

Reply:
Oh! But the fix is removing the quotes since DuckDB is insensitive, which seems correct.